python/cm/tests/test_api.py

```diff
@@ -0,0 +1,832 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Since this module is beyond QA responsibility we will not fix docstrings here
+# pylint: disable=missing-function-docstring, missing-class-docstring
+
+import os
+import string
+from uuid import uuid4
+
+from django.conf import settings
+from django.db import transaction
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework.test import APIClient
+
+from init_db import init as init_adcm
+from rbac.upgrade.role import init_roles
+
+
+class TestBase(TestCase):
+    files_dir = os.path.join(settings.BASE_DIR, "python", "cm", "tests", "files")
+
+    def setUp(self) -> None:
+        init_adcm()
+        init_roles()
+
+        self.client = APIClient(HTTP_USER_AGENT="Mozilla/5.0")
+        res = self.client.post(
+            path=reverse("rbac:token"),
+            data={"username": "admin", "password": "admin"},
+            format="json",
+        )
+        self.client.defaults["Authorization"] = f"Token {res.data['token']}"
+
+        self.client_unauthorized = APIClient(HTTP_USER_AGENT="Mozilla/5.0")
+
+        self.bundle_adh_name = "adh.1.5.tar"
+        self.bundle_ssh_name = "ssh.1.0.tar"
+
+    def load_bundle(self, bundle_name):
+        with open(os.path.join(self.files_dir, bundle_name), encoding="utf-8") as f:
+            with transaction.atomic():
+                response = self.client.post(
+                    path=reverse("upload-bundle"),
+                    data={"file": f},
+                )
+            self.assertEqual(response.status_code, 201, msg=response.content)
+        with transaction.atomic():
+            response = self.client.post(
+                path=reverse("load-bundle"),
+                data={"bundle_file": bundle_name},
+            )
+        self.assertEqual(response.status_code, 200, msg=response.content)
+
+
+class TestAPI(TestBase):  # pylint: disable=too-many-public-methods
+    cluster = "adh42"
+    host = "test.host.net"
+    service = "ZOOKEEPER"
+    component = "ZOOKEEPER_SERVER"
+
+    def get_service_proto_id(self):
+        response = self.client.get(reverse("service-type"))
+        self.assertEqual(response.status_code, 200, msg=response.content)
```

Comment:
It's better to use constantc from `rest_framework.statuses` for response codes. Assert message may help but in this case there can be attribute error instead of help

Reply:
done